### PR TITLE
11.0 [PORT] Sale Automatic Workflow

### PR DESCRIPTION
**Pull request by gdgellatly**
Hi,

I only did a really quick port here.  The only real changes were

- Update cron xml
- Change validate picking to use stock.immediate.transfer as do_transfer is deprecated so needed changing anyway.
- Fix some menus.
- Use widget="domain" instead of "char_domain"
- Bump version.

### CODE REVIEW

File: sale_automatic_workflow/data/automatic_workflow_data.xml
Comment (ghost) [inline sale_automatic_workflow/data/automatic_workflow_data.xml]: please put `name` before `ref`to be consistent w/ [these conventions](https://github.com/OCA/maintainer-tools/blob/master/CONTRIBUTING.md#format)

**Review by tschanzt — APPROVED** [review]

File: sale_automatic_workflow/i18n/sale_automatic_workflow.pot
Comment (ghost) [inline sale_automatic_workflow/i18n/sale_automatic_workflow.pot]: remove the file

File: sale_automatic_workflow/README.rst
Comment (ghost) [inline sale_automatic_workflow/README.rst]: 11.0

File: sale_automatic_workflow/README.rst
Comment (ghost) [inline sale_automatic_workflow/README.rst]: from [here](https://raw.githubusercontent.com/OCA/maintainer-tools/master/template/module/README.rst):
`help us smash it by providing detailed and welcomed feedback.`

File: sale_automatic_workflow/__manifest__.py
Comment (ghost) [inline sale_automatic_workflow/__manifest__.py]: aaand this one too

File: sale_automatic_workflow/__manifest__.py
Comment (ghost) [inline sale_automatic_workflow/__manifest__.py]: replace w/ `https://github.com/OCA/sale-workflow`

File: sale_automatic_workflow/__manifest__.py
Comment (ghost) [inline sale_automatic_workflow/__manifest__.py]: could you kindly format those properly? 😄

**Review by ghost — CHANGES_REQUESTED** [review]

File: sale_automatic_workflow/__init__.py
Comment (ghost) [inline sale_automatic_workflow/__init__.py]: replace `©` sign w/ `Copyright` :)

**Review by ghost — CHANGES_REQUESTED** [review]

File: sale_automatic_workflow/data/automatic_workflow_data.xml
Comment (gdgellatly) [reply sale_automatic_workflow/data/automatic_workflow_data.xml]: @arkostyuk that is not the convention for this particular field.  The convention is always ref first in OCA and Odoo core.  You can check the other thousands of matches to this regex `ref=.*name="model_id"`

File: sale_automatic_workflow/__manifest__.py
Comment (gdgellatly) [reply sale_automatic_workflow/__manifest__.py]: @arkostyuk While I agree with you that it is better, I do not think that is correct convention, traditionally we have not changed original authors websites.

File: sale_automatic_workflow/i18n/sale_automatic_workflow.pot
Comment (gdgellatly) [reply sale_automatic_workflow/i18n/sale_automatic_workflow.pot]: I need a second opinion on this, it was just a part of the migration

File: sale_automatic_workflow/__init__.py
Comment (gdgellatly) [reply sale_automatic_workflow/__init__.py]: second opinion as well please, I didn't write it, and there is nothing saying it isn't OK.  In all there are 27 of them.

File: sale_automatic_workflow/__manifest__.py
Comment (gdgellatly) [reply sale_automatic_workflow/__manifest__.py]: what do you mean exactly? The formatting is fine and passing flake8

**Review by atchuthan — APPROVED** [review]
Code review :+1:

File: sale_automatic_workflow/__manifest__.py
Comment (ghost) [reply sale_automatic_workflow/__manifest__.py]: AFAIR, the [convention](https://github.com/OCA/maintainer-tools/blob/master/CONTRIBUTING.md#modules) says that the `'website'` attribute should point to OCA repo for the sake of easing the access to the most relevant info about that module, though all of the author's credits (including website, I assume) shall remain in module's README :)

Comment (gdgellatly) [reply sale_automatic_workflow/__manifest__.py]: @arkostyuk interesting, never seen that before.  Usually we try and change as little as possible on forward ports, but I'll change this one but I'll leave to you the PR's for prior versions

File: sale_automatic_workflow/i18n/sale_automatic_workflow.pot
Comment (simahawk) [reply sale_automatic_workflow/i18n/sale_automatic_workflow.pot]: @gdgellatly you can wait for Pedro to tell you this :smile: 
Jokes apart - I dislike this - but we do drop POT files. Cannot find where all this started at the moment.

File: sale_automatic_workflow/views/sale_workflow_process_view.xml
Comment (simahawk) [inline sale_automatic_workflow/views/sale_workflow_process_view.xml]: to be removed?

File: sale_automatic_workflow/views/sale_workflow_process_view.xml
Comment (simahawk) [inline sale_automatic_workflow/views/sale_workflow_process_view.xml]: same?

**Review by simahawk — CHANGES_REQUESTED** [review]
LGTM but please squash translation commits together.
The are also 2 commented lines to be removed (or to be explained ;))

File: sale_automatic_workflow/__manifest__.py
Comment (ghost) [reply sale_automatic_workflow/__manifest__.py]: I was talking about putting each dependency on its own line, that makes changes clearer if a list of dependencies changes for whatever reason
For example,
```python
'depends': [
    'sale_stock',
    'sales_team',
],
```
and the `data` list may change even more often
This is fine enough though :)

File: sale_automatic_workflow/data/automatic_workflow_data.xml
Comment (ghost) [reply sale_automatic_workflow/data/automatic_workflow_data.xml]: that's interesting
In Odoo core:
`\bref=['"].*\bname=['"]model_id['"]`: 22 matches
`\bname=['"]model_id['"].*\bref=['"]`: 305 matches
so I'm still unsure :)
could you share a document having that convention in it?

**Review by ghost — APPROVED** [review]
LGTM

**Review by simahawk — APPROVED** [review]

**Review by hugosantosred — APPROVED** [review]
LGTM

**Pull request merged**
